xbmc/guilib/GUIAudioManager.cpp

```diff
@@ -25,6 +25,13 @@
 #include "utils/XBMCTinyXML.h"
 #include "utils/log.h"
 
+void CGUIAudioManager::IAESoundDeleter::operator()(IAESound* s)
+{
+  IAE* ae = CServiceBroker::GetActiveAE();
+  if (ae)
+    ae->FreeSound(s);
+}
```

Comment:
one more question ;)
is there a specific reason you chose a functor over a lambda for the deleter?

Reply:
When starting I initially thought I may need to create new `std::shared_ptr` in more then one place and using in place lambdas would have required to copy code. But this wasn't the case and I don't really have a preference so the functor stayed :)